Assets/Dependencies/SmashBrew/Character/States/MovementState.cs

```diff
@@ -187,14 +189,31 @@ struct MovementInfo {
             bool success = (!IsCrounching && JumpCount > 0 && JumpCount <= MaxJumpCount
                 && (Input.GetKeyDown(KeyCode.W) || Input.GetKeyDown(KeyCode.UpArrow)));
             if (success) {
-                CurrentLedge = null;
+                if (CurrentLedge != null)
+                    LeaveLedge();
                 OnJump.SafeInvoke();
                 PhysicsState.SetVerticalVelocity(_jumpPower[MaxJumpCount - JumpCount]);
                 CmdJump();
             }
             return success;
         }
 
+        void LeaveLedge() {
+            if (CurrentLedge == null)
+                return;
+            StageLedge ledge = CurrentLedge.GetComponent<StageLedge>();
+            if (ledge == null)
+                return;
+            CmdReleaseLedge();
+        }
+
+        public void tryLedgeGrab(StageLedge ledge) {
```

Comment:
Our code style specifies that methods should be capitalized.

Usually in C#, a method that is marked `TryX` returns a boolean saying whether the action succeeded or not.